basic-algorithms-&-datastructures/index100.js

```diff
@@ -0,0 +1,142 @@
+        username: "Mr. Coffee",
+        name: "Pat Comb"
+      },
+      likes: 10571,
+      dislikes: 14,
+      shares: 54,
+      comments: 461,
+      section: {
+        name: "Programming",
+        subscribers: 18101
+      },
+      url: "http://www.binary-bugle.com/programming-expectations-vs-reality"
+    },
+    {
+      headline: "Will Web Development Ever Rule the World?",
+      image: [],
+      date_published: "2018-12-29T10:00:00+09:00",
+      date_modified: "",
+      author: {
+        username: "webly_weberson",
+        name: "Julie Holmes"
+      },
+      likes: 8945,
+      dislikes: 31,
+      shares: 2,
+      comments: 109,
+      section: {
+        name: "Web Development",
+        subscribers: 9183
+      },
+      url:
+        "http://www.binary-bugle.com/will-web-development-ever-rule-the-world"
+    },
+    {
+      headline: "The Next Big Thing in Algorithms",
+      image: [],
+      date_published: "2018-12-30T11:58:00+11:00",
+      date_modified: "",
+      author: {
+        username: "enigmaGadget",
+        name: "Steve Ramsey"
+      },
+      likes: 459,
+      dislikes: 2,
+      shares: 1,
+      comments: 64,
+      section: {
+        name: "Algorithms",
+        subscribers: 5743
+      },
+      url: "http://www.binary-bugle.com/the-next-big-thing-in-algorithms"
+    },
+    {
+      headline: "Programming Explained in Fewer than 140 Characters",
+      image: [],
+      date_published: "2019-03-30T05:30:00+03:00",
+      date_modified: "",
+      author: {
+        username: "Mr. Coffee",
+        name: "Pat Comb"
+      },
+      likes: 308,
+      dislikes: 0,
+      shares: 3,
+      comments: 29,
+      section: {
+        name: "Programming",
+        subscribers: 18101
+      },
+      url:
+        "http://www.binary-bugle.com/programming-explained-in-fewer-than-140-characters"
+    },
+    {
+      headline: "This Week's Top Stories About Web Development",
+      image: [],
+      date_published: "2019-04-02T12:30:00+08:00",
+      date_modified: "",
+      author: {
+        username: "webly_weberson",
+        name: "Julie Holmes"
+      },
+      likes: 127,
+      dislikes: 10,
+      shares: 1,
+      comments: 34,
+      section: {
+        name: "Web Development",
+        subscribers: 9183
+      },
+      url:
+        "http://www.binary-bugle.com/this-weeks-top-stories-about-web-development"
+    },
+    {
+      headline: "Everything You've Ever Wanted to Know About A.I.",
+      image: [],
+      date_published: "2019-03-09T10:10:00+10:00",
+      date_modified: "",
+      author: {
+        username: "notarobot",
+        name: "Trista Campbell"
+      },
+      likes: 99,
+      dislikes: 3,
+      shares: 2,
+      comments: 31,
+      section: {
+        name: "Artificial Intelligence",
+        subscribers: 2374
+      },
+      url:
+        "http://www.binary-bugle.com/everything-youve-ever-wanted-to-know-about-ai"
+    }
+  ]
+};
+
+let newsFeedArray = []
+
+/*
```

Comment:
This is a very good explanation, but I believe the example code is missing the last curly brace. Here's a quick edit with an updated variable name:

/*
The `for...in` statement iterates over the properties of an object keyed by strings. For example: 

```js
const objExample = {animals: [{cats: 3}, {dogs: 4}, {birds: 7}]}

let animalsArray = [];

for (props in objExample) {
  animalsArray = objExample[props];
}
```

Convert our object newsFeed into an array by iterating over it with the `for...in` statement
and returning the result of each iteration into our newsFeedArray array. 
*/

Also, sorry, I believe I missed this in one of your earlier prototypes, but would it be easier to do the following?: `const newsFeedArray = newsFeed.articles;`